receiver/oracledbreceiver/scraper.go

```diff
@@ -430,6 +497,248 @@ func (s *oracleScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
+	s.logger.Debug("Metric columns", zap.Strings("names", metricNames))
+	s.logger.Debug("Cache", zap.Int("size", s.metricCache.Len()))
+	var hits []queryMetricCacheHit
+	var cacheUpdates, discardedHits int
+	for _, row := range metricRows {
+		newCacheVal := make(map[string]int64, len(metricNames))
+		for _, columnName := range metricNames {
+			val := row[columnName]
+			valInt64, err := strconv.ParseInt(val, 10, 64)
+			if err != nil {
+				errs = append(errs, err)
+			} else {
+				newCacheVal[columnName] = valInt64
+			}
+		}
+
+		cacheKey := fmt.Sprintf("%v:%v", row[sqlIDAttr], row[childNumberAttr])
+		// if we have a cache hit and the query doesn't belong to top N, cache is updated anyway
+		// as a result, once it finally makes its way to the top N queries, only the latest delta will be sent downstream
+		if oldCacheVal, ok := s.metricCache.Get(cacheKey); ok {
+			hit := queryMetricCacheHit{
+				sqlID:        row[sqlIDAttr],
+				queryText:    row[sqlTextAttr],
+				childNumber:  row[childNumberAttr],
+				childAddress: row[childAddressAttr],
+				metrics:      make(map[string]int64, len(metricNames)),
+			}
+
+			// it is possible we get a record with all deltas equal to zero. we don't want to process it any further
+			var possiblePurge, positiveDelta bool
+			for _, columnName := range metricNames {
+				delta := newCacheVal[columnName] - oldCacheVal[columnName]
+
+				// if any of the deltas is less than zero, cursor was likely purged from the shared pool
+				if delta < 0 {
+					possiblePurge = true
+					break
+				} else if delta > 0 {
+					positiveDelta = true
+				}
+
+				hit.metrics[columnName] = delta
+			}
+
+			// skip if possible purge or all the deltas are equal to zero
+			if !possiblePurge && positiveDelta {
+				hits = append(hits, hit)
+			} else {
+				discardedHits++
+			}
+		}
+		s.metricCache.Add(cacheKey, newCacheVal)
+		cacheUpdates++
+	}
+
+	// if cache updates is not equal to rows returned, that indicates there is problem somewhere
+	s.logger.Debug("Cache update", zap.Int("update-count", cacheUpdates), zap.Int("new-size", s.metricCache.Len()))
+
+	if len(hits) == 0 {
+		s.logger.Info("No log records for this scrape")
+		return plog.NewLogs(), errors.Join(errs...)
+	}
+
+	s.logger.Debug("Cache hits", zap.Int("hit-count", len(hits)), zap.Int("discarded-hit-count", discardedHits))
+	for _, hit := range hits {
+		s.logger.Debug(fmt.Sprintf("Cache hit, SQL_ID: %v, CHILD_NUMBER: %v", hit.sqlID, hit.childNumber), zap.Int64("elapsed-time", hit.metrics[elapsedTimeMetric]))
+	}
+
+	// order by elapsed time delta, descending
+	sort.Slice(hits, func(i, j int) bool {
+		return hits[i].metrics[elapsedTimeMetric] > hits[j].metrics[elapsedTimeMetric]
+	})
+
+	for _, hit := range hits {
+		s.logger.Debug(fmt.Sprintf("Cache hit after sorting, SQL_ID: %v, CHILD_NUMBER: %v", hit.sqlID, hit.childNumber), zap.String("child-address", hit.childAddress), zap.Int64("elapsed-time", hit.metrics[elapsedTimeMetric]))
```

Comment:
Too many debug statement makes the code harder to read. I believe this particular one can be extremely noisy since the hits list can be pretty big. Please try to maintain only valuable statements.